### ISSUE TITLE
hmr being included in build

### ISSUE THREAD

**Issue by DanPete** [hayes0724/shopify-webpack-themekit#6]
I'm having an issue where it seems that hmr files are being included in the `vendors@layout.theme.js` when I build. Then if I deploy the theme there are many error related to hmr not being included.

I found that if I removed this part of the code in `parts/core.js` then hot-client.js won't be included in the build vendor build.

 ```
Object.keys(core.entry).forEach((name) => {
    core.entry[name] = [
        path.join(__dirname, '../../hot-client.js')
    ].concat(core.entry[name])
})
```
But that also makes hmr reloading not work in dev, so is there another solution to prevent hmr from getting included in the build?

**Comment by jonashcroft** [comment]
i'm having the same issue. It looks like some of the config from `parts/core.js` should instead be in `lib/webpack/dev.config.js`.

What I did was remove that `Object.keys()` loop from `parts/core.js` and place it in `lib/webpack/dev.config.js`.

It's also likely that wrapping an `if (process.env.NODE_ENV !== production)` statement around the loop in `parts/core.js` could resolve this, but I haven't tried that.

**Comment by lizsterine** [comment]
> It's also likely that wrapping an if (process.env.NODE_ENV !== production) statement around the loop in parts/core.js could resolve this, but I haven't tried that.

Can confirm, that works too.  I'm doing this for my production builds.

**Comment by hayes0724** [comment]
This is fixed in 667221e9012715ec4af82cfc5017551448f97646, I moved the HMR code from core.js to dev.config.js
